### ISSUE TITLE
Settle on a per-corner rounding impl

### ISSUE THREAD

**Issue by WillPower3309** [wlrfx/scenefx#72]
Right now we have two different ways to round specific corners. One is used in rects where we have a different shader for each corner, and the other is in the tex shader where we do some `step` magic. We should settle on one impl for all corners.

One alternative idea I have is to completely remove the round_rect shader and the corner rounding logic in the tex shader, and damage everything EXCEPT the corners. Then we could damage the corners and have a shader for each corner. We'd need to do some benchmarking here, because this could either improve or decrease perf depending on how expensive damaging is vs saving the extra GPU instructions in the tex and rect shaders would be

**Comment by WillPower3309** [comment]
related to #70

**Comment by ErikReider** [comment]
> One alternative idea I have is to completely remove the round_rect shader and the corner rounding logic in the tex shader, and damage everything EXCEPT the corners. Then we could damage the corners and have a shader for each corner.

So it's either increase the CPU calls or increase the work of the GPU right. My gut feeling would be to go with the method we have now, but let's benchmark this! :D

**Comment by WillPower3309** [comment]
> > One alternative idea I have is to completely remove the round_rect shader and the corner rounding logic in the tex shader, and damage everything EXCEPT the corners. Then we could damage the corners and have a shader for each corner.
> 
> So it's either increase the CPU calls or increase the work of the GPU right. My gut feeling would be to go with the method we have now, but let's benchmark this! :D

Exactly! At the end of the day it'd probably be so minimal that the simplest option is the best one, which is likely what we do now

**Comment by WillPower3309** [comment]
At the end of the day this is primarily tracking the rect rounding impl matching the tex one

**Comment by ErikReider** [comment]
Can we close this?
